ScatterGather/ScatterGather/ViewController.swift

```diff
@@ -0,0 +1,178 @@
+                    }
+                }
+                UIView.addKeyframe(withRelativeStartTime: 0.4, relativeDuration: 0.2) {
+                    for label in self.letterLabels {
+                        let rotationAngle = CGFloat(Int.random(in: 1...20)) / 10
+                        label.transform = CGAffineTransform(rotationAngle: rotationAngle * .pi)
+                        label.center.x = CGFloat(Int.random(in: 0...Int(self.view.bounds.size.width)))
+                        label.center.y = CGFloat(Int.random(in: Int(self.view.bounds.size.height * 0.4)...Int(self.view.bounds.size.height * 0.6)))
+                    }
+                }
+                UIView.addKeyframe(withRelativeStartTime: 0.6, relativeDuration: 0.2) {
+                    for label in self.letterLabels {
+                        let rotationAngle = CGFloat(Int.random(in: 1...20)) / 10
+                        label.transform = CGAffineTransform(rotationAngle: rotationAngle * .pi)
+                        label.center.x = CGFloat(Int.random(in: 0...Int(self.view.bounds.size.width)))
+                        label.center.y = CGFloat(Int.random(in: Int(self.view.bounds.size.height * 0.6)...Int(self.view.bounds.size.height * 0.8)))
+                    }
+                }
+                UIView.addKeyframe(withRelativeStartTime: 0.8, relativeDuration: 0.2) {
+                    self.currentRotationValues = []
+                    for label in self.letterLabels {
+                        let newRotationAngle = (CGFloat(Int.random(in: 1...20)) / 10) * .pi
+                        self.currentRotationValues.append(newRotationAngle)
+                        label.transform = CGAffineTransform(rotationAngle: newRotationAngle)
+                        label.center.x = CGFloat(Int.random(in: 0...Int(self.view.bounds.size.width)))
+                        label.center.y = CGFloat(Int.random(in: Int(self.view.bounds.size.height * 0.8)...Int(self.view.bounds.size.height)))
+                    }
+                }
+                
+            }
+            
+            UIView.animateKeyframes(withDuration: 4, delay: 0, options: [], animations: animBlock, completion: nil)
+            
+            isScattered = true
+            
+        case true:
+            UIView.animate(withDuration: 2) {
+                self.logoImageView.alpha = 1
+                for label in self.letterLabels {
+                    label.layer.backgroundColor = UIColor(hue: 349/359, saturation: 0.90, brightness: 0.73, alpha: 1).cgColor
+                    label.textColor = UIColor(hue: 180/359, saturation: 0, brightness: 1, alpha: 1)
+                    label.shadowColor = UIColor(hue: 349/359, saturation: 1, brightness: 0.5, alpha: 1)
+                }
+            }
+            self.takeYourPositions()
+            isScattered = false
+        }
```

Comment:
I would think about refactoring this into two functions, one for scattering and one for gathering.